Core/include/Acts/Material/PropagatorMaterialAssigner.hpp

```diff
@@ -0,0 +1,164 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/EventData/TrackParameters.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/MagneticField/MagneticFieldContext.hpp"
+#include "Acts/Material/interface/IAssignmentFinder.hpp"
+#include "Acts/Propagator/AbortList.hpp"
+#include "Acts/Propagator/ActionList.hpp"
+#include "Acts/Propagator/Propagator.hpp"
+#include "Acts/Propagator/SurfaceCollector.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+#include "Acts/Utilities/Logger.hpp"
+#include "Acts/Utilities/VectorHelpers.hpp"
+
+#include <utility>
+#include <vector>
+
+namespace Acts {
+
+/// @brief selector for finding surface
+struct MaterialSurface {
```

Comment:
```suggestion
struct MaterialSurfaceSelector {
```

Reply:
I renamed it into `MaterialSurfaceIdentifier` because it is to be used with the already existing 
`SurfaceSelector<>` as 

```
using MaterialSurfaceSelector = SurfaceSelector< MaterialSurfaceIdentifier >
```

which then fullfills the `ActionList` API.